gocd/gocd_test.go

```diff
@@ -61,8 +63,52 @@ func TestGocdPOST(t *testing.T) {
 
 func TestGocdGET(t *testing.T) {
 	fmt.Println("TestGocdGET")
-	_, err := pipelineGET("http://localhost:8153", "test")
+	_, _, err := pipelineGET("http://localhost:8153", "test")
 	if err != nil {
 		t.Error(err)
 	}
 }
+
+func TestExist(t *testing.T) {
+	fmt.Println("TestExist")
+	etag := Exist("http://localhost:8153", "test")
+	if etag == "" {
+		t.Error("test does not exist as a gocd pipeline")
+	}
+}
+
+func TestGocdPUT(t *testing.T) {
+	fmt.Println("TestGocdPUT")
+	pipeline, etag, _ := pipelineGET("http://localhost:8153", "test")
+
+	strangeIndex := sort.Search(len(pipeline.EnvironmentVariables), func(index int) bool { return pipeline.EnvironmentVariables[index].Name == "STRANGE" })
```

Comment:
sort.Search actually does a binary search, assumes the elements "EnvironmentVariables" are sorted by the key I am checking. Must rework to use a for loop. Will comment on the changing nature of the newly edited pipelines

https://golang.org/pkg/sort/#Search